apps/api/v2/src/modules/billing/services/billing.service.ts

```diff
@@ -99,43 +102,49 @@ export class BillingService {
     );
   }
 
-  async increaseUsageForTeam(teamId: number) {
-    try {
-      const billingSubscription = await this.billingRepository.getBillingForTeam(teamId);
-      if (!billingSubscription || !billingSubscription?.subscriptionId) {
-        this.logger.error("Team did not have stripe subscription associated to it", {
-          teamId,
-        });
-        return void 0;
-      }
-
-      const stripeSubscription = await this.stripeService.stripe.subscriptions.retrieve(
-        billingSubscription.subscriptionId
-      );
-      const item = stripeSubscription.items.data[0];
-      // legacy plans are licensed, we cannot create usage records against them
-      if (item.price?.recurring?.usage_type === "licensed") {
-        return void 0;
-      }
+  /**
+   *
+   * Adds a job to the queue to increment usage of a stripe subscription.
+   * we delay the job until the booking starts.
+   * the delay ensure we can adapt to cancel / reschedule.
+   */
+  async increaseUsageByClientId(
+    clientId: string,
+    booking: {
+      uid: string;
+      startTime: Date;
+      fromReschedule?: string | null;
+    }
+  ) {
+    const { uid, startTime, fromReschedule } = booking;
 
-      await this.stripeService.stripe.subscriptionItems.createUsageRecord(item.id, {
-        action: "increment",
-        quantity: 1,
-        timestamp: "now",
-      });
-    } catch (error) {
-      // don't fail the request, log it.
-      this.logger.error("Failed to increase usage for team", {
-        teamId: teamId,
-        error,
-      });
+    const delay = startTime.getTime() - Date.now();
+    if (fromReschedule) {
+      // cancel the usage increment job for the booking that is being rescheduled
+      await this.cancelUsageByBookingUid(fromReschedule);
+      this.logger.log(`Cancelled usage increment job for rescheduled booking uid: ${fromReschedule}`);
     }
+    await this.billingQueue.add(
+      INCREMENT_JOB,
+      {
+        oAuthClientId: clientId,
+      } satisfies IncrementJobDataType,
```

Comment:
I think we can skip using zod since we use satisfies here and this should be the only place we ever call the INCREMENT_JOB